components/List/docs/create-your-own.md

```diff
@@ -0,0 +1,208 @@
+
+```objc
+- (void)setHighlighted:(BOOL)highlighted {
+  [super setHighlighted:highlighted];
+  if (highlighted) {
+    [_inkView startTouchBeganAnimationAtPoint:_lastTouch completion:nil];
+  } else {
+    [_inkView startTouchEndedAnimationAtPoint:_lastTouch completion:nil];
+  }
+}
+```
+
+6. When the cell is reused we must make sure no outstanding ripple animations stay on the cell so we need to clear the ink before:
+
+```objc
+- (void)prepareForReuse {
+  [_inkView cancelAllAnimationsAnimated:NO];
+  [super prepareForReuse];
+}
+```
+
+Now there is ink in our cells!
+
+### Self Sizing
+In order to have cells self-size based on content and not rely on magic number constants to decide how big they should be, we need to follow these steps:
+
+1. apply autoulayout constraints of our added subviews relative to each other and their superview (the cell's `contentView`). We need to make sure our constraints don't define static heights or widths but rather constraints that are relative or our cell won't calculate itself based on the dynamically sized content.
+
+You can see how it is achieved in the `(void)setupConstraints` method in our example. If you'll notice there are some constraints that are set up to be accessible throughout the file:
+
+```objc
+NSLayoutConstraint *_imageLeftPaddingConstraint;
+NSLayoutConstraint *_imageRightPaddingConstraint;
+NSLayoutConstraint *_imageWidthConstraint;
+``` 
+This is in order to support the changing layout if an image is set or not.
+
+2. Because our list cells need to fill the entire width of the collection view, we want to expose the cell's width to be settable by the view controller when the cell is set up. For that we expose a `setCellWidth` method that sets the width constraint of the `contentView`:
+
+```objc
+- (void)setCellWidth:(CGFloat)width {
+  _cellWidthConstraint.constant = width;
+  _cellWidthConstraint.active = YES;
+}
+```
+
+and then in the collection view's `cellForItemAtIndexPath` delegate method we set the width:
+
+```objc
+CGFloat cellWidth = CGRectGetWidth(collectionView.bounds);
+#if defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
+if (@available(iOS 11.0, *)) {
+  cellWidth -=
+    (collectionView.adjustedContentInset.left + collectionView.adjustedContentInset.right);
+}
+#endif
+[cell setCellWidth:cellWidth];
+```
+
+
+3. In our collection view's flow layout we must set an `estimatedItemSize` so the collection view will defer the size calculations to its content.
+
+Note: It is better to set the size smaller rather than larger or constraints might break in runtime.
+
+```objc
+_flowLayout.estimatedItemSize = CGSizeMake(kSmallArbitraryCellWidth, kSmallestCellHeight);
+```
+
+### Typography
+
+For our example we use a typography scheme to apply the fonts to our cell's `UILabel`'s. Please see <a href="../schemes/Typography">Typography Scheme</a> for more info.
```

Comment:
This likely needs to be `../../schemes/Typography` now that we're in the docs folder.

Reply:
Thanks!